Sources/SwiftParser/Diagnostics/MissingNodesError.swift

```diff
@@ -233,19 +260,30 @@ extension ParseDiagnosticsGenerator {
       return .skipChildren
     }
 
+    /// Ancestors that don't contain any tokens are not very interesting to merge diagnostics (because there can't be any missing tokens we can merge them with).
+    /// Find the first ancestor that contains any tokens.
+    var ancestorWithTokens = node.parent
+    var index = node.index
+    while let unwrappedParent = ancestorWithTokens, !unwrappedParent.hasTokens {
+      ancestorWithTokens = unwrappedParent.parent
+      index = unwrappedParent.index
+    }
```

Comment:
This is almost `ancestorOrSelf { $0.hasTokens }` but it gives the index of the child that you came from. Could add that to `ancestorOrSelf`?

Reply:
I thought about it but decided against it because I think the index is not really useful in any other place and it would make `ancestorOfSelf` harder to use (you‘d need to extract a value from the tuple `ancestorOrSelf` would now return).